src/v2/Apps/Partner/Components/PartnerArtists/PartnerArtistsCarousel/PartnerArtistsCarousel.tsx

```diff
@@ -0,0 +1,190 @@
+import {
+  Box,
+  Carousel,
+  CarouselCell,
+  Text,
+  Flex,
+  Swiper,
+  CarouselProps,
+} from "@artsy/palette"
+import { flatten } from "lodash"
+import React from "react"
+import { createFragmentContainer, graphql, QueryRenderer } from "react-relay"
+import { useSystemContext } from "v2/Artsy"
+import { RouterLink } from "v2/Artsy/Router/RouterLink"
+import { Media } from "v2/Utils/Responsive"
+import { PartnerArtistsCarouselQuery } from "v2/__generated__/PartnerArtistsCarouselQuery.graphql"
+import { PartnerArtistsCarousel_partner } from "v2/__generated__/PartnerArtistsCarousel_partner.graphql"
+import {
+  PartnerArtistsCarouselItemFragmentContainer,
+  ResponsiveImage,
+} from "./PartnerArtistsCarouselItem"
+import { PartnerArtistsCarouselPlaceholder } from "./PartnerArtistsCarouselPlaceholder"
+
+const PAGE_SIZE = 19
+
+const DesktopCarousel: React.FC<
+  CarouselProps & { itemsPerViewport: number }
+> = ({ children, itemsPerViewport, ...rest }) => {
+  return (
+    <Carousel
+      Cell={React.forwardRef((props, ref) => {
+        return (
+          <CarouselCell
+            display="inline-flex"
+            flexGrow={0}
+            flexShrink={0}
+            style={{
+              boxSizing: "initial",
+              flexBasis: `calc((100% - ${
+                itemsPerViewport - 1
+              } * 20px) / ${itemsPerViewport})`,
+            }}
+            {...props}
+            ref={ref as any}
+          />
+        )
+      })}
+      {...rest}
+    >
+      {children}
+    </Carousel>
+  )
+}
+
+export interface PartnerArtistsCarouselProps {
+  partner: PartnerArtistsCarousel_partner
+}
+
+export interface PartnerArtistsCarouselContainerProps {
+  children: (itemsPerViewport: number) => JSX.Element | JSX.Element[]
+}
+
+export const PartnerArtistsCarouselContainer: React.FC<PartnerArtistsCarouselContainerProps> = ({
+  children,
+}) => {
+  return (
+    <>
+      <Media greaterThan="md">
+        <DesktopCarousel itemsPerViewport={4}>{children(4)}</DesktopCarousel>
+      </Media>
+      <Media at="md">
+        <DesktopCarousel itemsPerViewport={3}>{children(3)}</DesktopCarousel>
+      </Media>
+      <Media at="sm">
+        <DesktopCarousel itemsPerViewport={2}>{children(2)}</DesktopCarousel>
+      </Media>
+      <Media at="xs">
+        <Swiper>{children(2)}</Swiper>
+      </Media>
+    </>
+  )
+}
```

Comment:
Looks good — but I'd push back a little on design next time if we have to jump through this many hoops to make something work. We should be following patterns which are less complicated.